extensions/barrage/src/main/java/io/deephaven/extensions/barrage/BarrageSnapshotOptions.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+
+package io.deephaven.extensions.barrage;
+
+import com.google.flatbuffers.FlatBufferBuilder;
+import io.deephaven.annotations.BuildableStyle;
+import io.deephaven.barrage.flatbuf.BarrageSnapshotRequest;
+import org.immutables.value.Value.Default;
+import org.immutables.value.Value.Immutable;
+
+@Immutable
+@BuildableStyle
+public abstract class BarrageSnapshotOptions {
+    public static Builder builder() {
+        return ImmutableBarrageSnapshotOptions.builder();
+    }
+
+    public static BarrageSnapshotOptions of(final io.deephaven.barrage.flatbuf.BarrageSnapshotOptions options) {
+        if (options == null) {
+            return builder().build();
+        }
+        final byte mode = options.columnConversionMode();
+        return builder()
+                .useDeephavenNulls(options.useDeephavenNulls())
+                .columnConversionMode(conversionModeFbToEnum(mode))
+                .batchSize(options.batchSize())
+                .build();
+    }
+
+    public static BarrageSnapshotOptions of(final BarrageSnapshotRequest snapshotRequest) {
+        return of(snapshotRequest.snapshotOptions());
+    }
+
+    /**
+     * By default, prefer to communicate null values using the arrow-compatible validity structure.
+     *
+     * @return whether to use deephaven nulls
+     */
+    @Default
+    public boolean useDeephavenNulls() {
+        return false;
+    }
+
+    /**
+     * @return the preferred batch size if specified
+     */
+    @Default
+    public int batchSize() {
+        return 0;
+    }
+
+    @Default
+    public ColumnConversionMode columnConversionMode() {
+        return ColumnConversionMode.Stringify;
+    }
+
+    public int appendTo(FlatBufferBuilder builder) {
+        return io.deephaven.barrage.flatbuf.BarrageSnapshotOptions.createBarrageSnapshotOptions(
+                builder, conversionModeEnumToFb(columnConversionMode()), useDeephavenNulls(), batchSize());
+    }
+
+    private static ColumnConversionMode conversionModeFbToEnum(final byte mode) {
```

Comment:
Please migrate these methods from `BarrageSnapshotOptions` and `BarrageSubscriptionOptions` to `ColumnConversionMode` and make them package-private.